### ISSUE TITLE
kernel panic when kill user process

### ISSUE THREAD

**Issue by JaybeamChinghw** [cryptodev-linux/cryptodev-linux#33]
I am using cryptodev to access linux kernel cryptographic algorithm driver, if the data source address and destination address of encryption or decryption was different, and I killed the userspace testing process with a command just like "kill -s 9" or "kill -s 15", then kernel panic occured.

**Comment by cristian-stoica** [comment]
What AES (from the panic) test did you run and what kernel algortithm(s) did you enabled? What are the versions of kernel and cryptodev?

**Comment by JaybeamChinghw** [comment]
aes_test was a testing program of user space, and it was coded after the file examples/aes.c. I put  the three functions: aes_ctx_init， aes_encrypt，aes_ctx_deinit into a while loop for a longtime test, and killed the aes_test while it was running. Kernel algortithm was xts(aes), and I just added several lines to crypto_create_session to support this algortithm. The version of cryptodev is 1.9, and kernel is 4.1.46. The               
 kernel algortithm(s) I enabled were: stdrng, lzo, crct10dif, crc32c, ecb(arc4), arc4, aes, des3_ede, des, sha224, sha256, sha1, md5, md4.

**Comment by cristian-stoica** [comment]
Did you get the crash also with vanilla examples/aes.c or with some of the tests? You could modify them to run a little longer for an easy time to CTRL-C.
The kernel algorithms question was to see if you run a software or a hardware AES driver - it is irrelevant though since the crash is on a different code path.

**Comment by JaybeamChinghw** [comment]
I just used the branch you gave to test again, and modified  examples/aes.c, made the source address and destination address different in the aes_encrypt function. I tested kill -9 or -15 while running for several times,  the crash always happened. Now I'm not sure the problem belongs to cryptodev or the kernel, I will change the kernel version to test it again, and maybe tomorrow I could give you the result. Thank you very much!

**Comment by cristian-stoica** [comment]
If I understand you correctly you always get the crash with different destination buffer but not with in-place operations?

**Comment by JaybeamChinghw** [comment]
Yes, you have gotten the point, the key point is "different destination buffer", and no crash happened with in-place operations.
So sorry to have bother you, wish you have a happy vacation.
Thanks!

**Comment by JaybeamChing** [comment]
This week , I did the same test with other kernel version——4.10.0-42-generic，and it also reported calltrace information as following:
[10420.249240] [<ffff000001c3ea48>] release_user_pages+0x90/0x100 [cryptodev]
[10420.249246] [<ffff000001c3ece0>] get_userbuf+0x228/0x310 [cryptodev]
[10420.249252] [<ffff000001c3b7c8>] crypto_run+0x1c8/0x4a8 [cryptodev]
[10420.249257] [<ffff000001c3ab88>] cryptodev_ioctl+0x520/0x7d8 [cryptodev]
[10420.249260] [<ffff0000082bd52c>] do_vfs_ioctl+0xc4/0x8e8
[10420.249262] [<ffff0000082bdde4>] SyS_ioctl+0x94/0xa8
[10420.249264] [<ffff0000080838f0>] el0_svc_naked+0x24/0x28
[10420.249266] Code: 52800080 d37df035 f8617841 8b150042 (f9400023) 
[10420.249268] ---[ end trace a5d9b3b46941a846 ]---

So I think this problem  has nothing with kernel version. I will be glad to research it with you when you return to work.

**Comment by cristian-stoica** [comment]
@JaybeamChing Am I wrong to ask if you've got a patch for this?

**Comment by JaybeamChing** [comment]
Not yet, I am still not sure how to solve this problem.

**Comment by JaybeamChing** [comment]
Do you have any idea how to fix this bug now?

**Comment by cristian-stoica** [comment]
@JaybeamChing I'm sorry I don't have a solution for you. I am currently involved in other projects and not actively debugging this issue.
